src/PostgREST/Plan.hs

```diff
@@ -137,51 +126,57 @@ treeRestrictRange maxRows _ request = pure $ nodeRestrictRange maxRows <$> reque
     nodeRestrictRange :: Maybe Integer -> ReadPlan -> ReadPlan
     nodeRestrictRange m q@ReadPlan{range_=r} = q{range_=restrictRange m r }
 
-augmentRequestWithJoin :: Schema -> RelationshipsMap -> ReadPlanTree -> Either ApiRequestError ReadPlanTree
-augmentRequestWithJoin schema allRels request =
- addJoinConditions Nothing <$> addRels schema allRels Nothing request
-
-addRels :: Schema -> RelationshipsMap -> Maybe ReadPlanTree -> ReadPlanTree -> Either ApiRequestError ReadPlanTree
-addRels schema allRels parentNode (Node query@ReadPlan{from=tbl,nodeName,nodeHint,nodeDepth} forest) =
+-- add relationships to the nodes of the tree by traversing the forest while keeping track of the parentNode, also adds aliasing
+addRels :: Schema -> Action -> RelationshipsMap -> Maybe ReadPlanTree -> ReadPlanTree -> Either ApiRequestError ReadPlanTree
+addRels schema action allRels parentNode (Node rPlan@ReadPlan{relName,relHint,depth} forest) =
   case parentNode of
-    Just (Node ReadPlan{from=parentNodeQi, fromAlias=aliasQi} _) ->
-      let newFrom r = if qiName tbl == nodeName then relForeignTable r else tbl
-          newReadPlan = (\r ->
-            if not $ relIsSelf r -- add alias if self rel TODO consolidate aliasing in another function
```

Comment:
It reduces to this. We used to create aliases only for self relationships, now we do it for all relationships.